src/EditorFeatures/CSharpTest/CommentSelection/ToggleBlockCommentCommandHandlerTests.cs

```diff
@@ -0,0 +1,1993 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CommentSelection;
+using Microsoft.CodeAnalysis.Editor.CSharp.CommentSelection;
+using Microsoft.CodeAnalysis.Editor.Implementation.CommentSelection;
+using Microsoft.CodeAnalysis.Editor.UnitTests;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Utilities;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Experiments;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Operations;
+using Roslyn.Test.Utilities;
+using Roslyn.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.CommentSelection
+{
+    [UseExportProvider]
+    public class ToggleBlockCommentCommandHandlerTests
```

Comment:
are vb tests somewhere?  i couldn't find any.

Reply:
VB doesn't have block comments

Reply:
I'm not certain why that's relevant *from the user's perspective*.  The user ask (afaict) is to have a "toggle comment" option.  So you can just go and do things like select code and say "toggle".  That would certainly work for both C# and VB for non-block-comment selections.

Reply:
In this case, the ask is for specifically a toggle *block* comment feature, similar to VScode, R#, etc.

It's potentially debatable if what they really want is a toggle comment feature, but we've already arrived separate commands to start with.

Reply:
> It's potentially debatable if what they really want is a toggle comment feature, but we've already arrived separate commands to start with.

This debate should happen.  i think it's possible people over-specialized on hte 'ask'.  

> the ask is for specifically a toggle block comment feature, similar to VScode, R#, etc.

Sure, i get that that was how people started thinking about it.  But we should ask if the real need is for separate commands.  Or just a a single toggle command.  But we can move to the linked design issue.

Reply:
This feature is Toggle Block Comment (and the command name explicitly states that the comment will be a *block comment*). There is no conceptual mapping to VB. We are not attempting to address a Toggle Line Comment feature with this change; if you want such a feature please feel free to open a separate issue for it.